iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -1181,120 +1176,100 @@ private void HandleTwinResponse(MqttApplicationMessageReceivedEventArgs received
             {
                 byte[] payloadBytes = receivedEventArgs.ApplicationMessage.Payload ?? Array.Empty<byte>();
 
-                if (_getTwinResponseCompletions.TryRemove(receivedRequestId, out TaskCompletionSource<GetTwinResponse> getTwinCompletion))
+                if (_pendingTwinOperations.TryRemove(receivedRequestId, out PendingMqttTwinOperation twinOperation))
                 {
                     if (Logging.IsEnabled)
-                        Logging.Info(this, $"Received response to get twin request with request id {receivedRequestId}.");
+                        Logging.Info(this, $"Received response to patch twin request with request id {receivedRequestId}.", nameof(HandleTwinResponse));
 
-                    if (status != 200)
+                    IotHubClientErrorResponseMessage ParseError(byte[] payloadBytes)
```

Comment:
I'm not a huge fan of functions within functions like this. Thoughts on moving it out and making it static?

Reply:
And rename to "ParseTwinResponseError" to be more specific

Reply:
I think having a small nested function is nice so it doesn't pollute the overall class API surface as this API is only used by this method.